user/how-to-guides/how-to-organize-your-qubes.md

```diff
@@ -0,0 +1,451 @@
+of like Lego blocks in the sense that you can build whatever you want. But if
+you're not sure what to build, then this open-ended freedom can be daunting.
+It's a bit like staring at a blank document when you first sit down to write
+something. The possibilities are endless, and you may not know where to begin!
+
+The truth is that no one else can tell you *exactly* how you should organize
+your qubes, as there is no single correct answer to that question. It depends
+on your needs, desires, and preferences. Every user's optimal setup will be
+different. However, what we *can* do is provide you with some illustrative
+examples based on questionnaires and interviews with Qubes users and
+developers, as well as our own personal experience and insight from using Qubes
+over the years. You may be able to adapt some of these examples to fit your own
+unique situation. More importantly, walking you through the rationale behind
+various decisions will teach you how to apply the same thought process to your
+own organizational decisions. Let's begin!
+
+
+## Alice, the software developer
+
+Alice is a freelance dev who works on several projects for different clients
+simultaneously. The projects have varying requirements and often different
+build environments. She has a separate set of qubes for each project. She keeps
+them organized by coming up with a naming scheme, such as:
+
+```
+client1-code
+client1-build
+client1-test
+client1-prod
+client2-code
+client2-build
+client2-test 
+client2-prod
+...
+```
+
+This helps her keep groups of qubes organized in a set. Some of her qubes are
+based on [Debian templates](/doc/templates/debian/), while others are based on
+[Fedora templates](/doc/templates/fedora/). The reason for this is that some
+software packages are more readily available in one distribution as opposed to
+the other. Alice's setup looks like this:
+
+- Several qubes for writing code. Here's where she runs her IDE, commits code,
+  and signs her commits. These qubes are based on different templates depending
+  on which tools and which development environment she needs. In general, Alice
+  likes to have a separate qube of this type for each client or each project.
+  This allows her to keep everything organized and avoid accidentally mixing up
+  any access credentials or client code, which could be disastrous. This also
+  allows her to truthfully tell her clients that their code is always securely
+  isolated from all her other clients. She likes to use the [Qubes
+  firewall](/doc/firewall/) to restrict these qubes' network access to only the
+  code repositories she needs in that qube in order to avoid accidentally
+  interacting with anything else on her local network or on the internet. Alice
+  also has some qubes of this type for personal programming projects that she
+  works on just for fun when she has "free time" (whatever that is).
+
+- Several qubes for building and testing. Again, Alice usually likes to have
```

Comment:
@marmarek , do you think this is a reasonable setup? It feels a bit cumbersome and RAM-intensive to me to have code/build/production qubes for each project separate.

Reply:
Full NxM matrix indeed is a bit extreme, but I know at least few people that do similar separation at least for some projects (and several more that separate just for each client).

Reply:
Any ideas for how to make the setup more realistic are welcome! This is admittedly speculation on my part, as I don't run such a setup myself but hang around people who do enough to get a vague sense of how you guys operate. :)

Reply:
I'm not sure if this is enough of an improvement, but I've attempted to revise this part a bit to acknowledge that maximum compartmentalization can sometimes be too cumbersome and memory-intensive, so Alice sometimes doesn't go that far when she decides it isn't worth it. Let me know if you have ideas for further improvement!